src/main/java/ladysnake/spawnercontrol/SpawnerEventHandler.java

```diff
@@ -211,6 +215,54 @@ public static void onLivingExperienceDrop(LivingExperienceDropEvent event) {
         }
     }
 
+    @SubscribeEvent(priority = EventPriority.LOWEST)
+    public static void onLivingItemDrop(LivingDropsEvent event) {
+        EntityLivingBase livingBase = event.getEntityLiving();
+        SpawnerConfig cfg = SpawnerUtil.getConfig(livingBase.getEntityData().getTag(NBT_TAG_SPAWNER_POS));
+        if (cfg != null) {
+            try {
+                ResourceLocation rl = EntityList.getKey(event.getEntity());
+                if (rl != null) {
+                    SpawnerConfig.MobLoot.MobLootEntry entry = cfg.mobLoot.lootEntries.get(rl);
+                    List<EntityItem> drops = event.getDrops();
+                    if (entry.removeAllItems) {
+                        drops.clear();
+                    }
+                    else {
+                        for (EntityItem drop : drops.toArray(new EntityItem[0])) {
+                            ItemStack stack = drop.getItem();
+                            for (String s : entry.removedItems) {
+                                String[] split = s.split(":");
+                                if (stack.getUnlocalizedName().equals(split[0] + ":" + split[1])
+                                        && (split.length < 3 || stack.getMetadata() == Integer.parseInt(split[2]))) {
+                                    drops.remove(drop);
+                                }
+                            }
+                        }
+                    }
+                    World world = livingBase.world;
+                    double x = livingBase.posX, y = livingBase.posY, z = livingBase.posZ;
+                    for (String s : entry.addedItems)
+                    {
+                        String[] split = s.split(":");
+                        if (split.length < 5 || world.rand.nextFloat() < Double.parseDouble(split[4])) {
```

Comment:
The comparison is inconsistent. Either use `rand.nextFloat() < Float.parseFloat()` or `rand.nextDouble() < Double.parseDouble()`, mixing the two loses the double precision while making it unclear which precision you are expecting.

Reply:
Woops, that was a copy-paste from the existing code; guess I'll fix both

Reply:
Oh. Well, fuck me.